src/main/java/core/basesyntax/strategy/ActionReaderStrategy.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Transaction;
+import core.basesyntax.service.CsvManagerImpl;
+import core.basesyntax.strategy.actions.ActionHandler;
+import core.basesyntax.strategy.actions.BaseActionHandler;
+import core.basesyntax.strategy.actions.PurchaseActionHandler;
+import core.basesyntax.strategy.actions.SupplyActionHandler;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ActionReaderStrategy {
+    private static final CsvManagerImpl mapHandler = new CsvManagerImpl();
+
+    public Map<String, Integer> inputDataToMap(String path) {
+        Map<String, Integer> stock = new HashMap<>();
+        List<Transaction> actionList = mapHandler.read(path);
+        actionList.forEach(a -> {
+            ActionHandler reportService;
+            String action = a.getAction();
+            String data = a.getData();
+            switch (action) {
+                case "b":
+                    reportService = new BaseActionHandler();
+                    break;
+                case "s":
+                case "r":
+                    reportService = new SupplyActionHandler();
+                    break;
+                case "p":
+                    reportService = new PurchaseActionHandler();
+                    break;
+                default:
+                    throw new RuntimeException(
+                            "Error within a file: Action is not specified correctly");
+            }
```

Comment:
Avoid using switch-cases and if-else constructions. It is recommended to use the Strategy pattern instead. In the main() method you can pass the strategy map into the service that chooses the correct strategy for each operation. see https://mate-academy.github.io/jv-program-common-mistakes/java-core/solid/fruit-shop